### PR TITLE
HDDS-4402. Recon dashboard page does not load until missing containers API returns data.

### PR DESCRIPTION
**Pull request by smitajoshi12**
What changes were proposed in this pull request?
Remove missingContainer Endpoint and use missingCounter from ClusterState End Point.
On a cluster with million containers, it was observed that the Recon dashboard page does not load even when most of the information has been returned by the backend through the /clusterState API, but the /missing containers API is still blocked.

What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-4402

How was this patch tested?
Manually Tested

image

### CODE REVIEW

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/db.json [hunk 1]
```diff
@@ -1,15 +1,15 @@
 {
   "clusterState": {
+    "pipelines": 32,
     "totalDatanodes": 24,
     "healthyDatanodes": 24,
-    "pipelines": 32,
     "storageReport": {
-      "capacity": 32985348833280,
-      "used": 15942918602752,
-      "remaining": 12094627905536
```
Comment (dombizita) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/db.json:9]: Can I ask why did you change this section?

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx [hunk 2]
```diff
@@ -90,14 +90,12 @@ export class Overview extends React.Component<Record<string, object>, IOverviewS
...
-      axios.get('/api/v1/containers/missing'),
       axios.get('/api/v1/task/status')
-    ]).then(axios.spread((clusterStateResponse, missingContainersResponse,taskstatusResponse) => {
+    ]).then(axios.spread((clusterStateResponse, taskstatusResponse) => {
       
       const clusterState: IClusterStateResponse = clusterStateResponse.data;
-      const missingContainers: IMissingContainersResponse = missingContainersResponse.data;     
       const taskStatus = taskstatusResponse.data;
-      const missingContainersCount = missingContainers.totalCount;
+      const missingContainersCount = clusterState.missingContainers;
```
Comment (dombizita) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:98]: Based on the implementation presented in #3952 if the `clusterState.missingContainers` is 1001, that means we have more than 1000 missing containers in the cluster. We should present this on the UI, if the count is 1001 we can show eg. `1000+` on the overview page and add some tooltip, that the user can access the exact number on the Containers page.

**Review by dombizita — COMMENTED** [review]
Thanks for the patch @smitajoshi12, overall it looks good to me, I added some inline comments, please take a look at them.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/db.json (hunk 1 above)
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/db.json:9]: I have copied actual end point Response in DB.json so it will be same.

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx (hunk 2 above)
Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:98]: image
It is existing Functionality Could you tell me exact tool tip message.

Comment (dombizita) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:98]: If the endpoint returns 1001 as the number of missing containers, that means that actually we have more than 1000 missing containers and to avoid long loading on the overview page we don't know the exact number. The tooltip should show something like this (if 1001 containers are missing): "1000+ containers are missing, for more information go to Containers page". If it is less than 1001, you can show the exact number instead of "1000+". What do you think? @smitajoshi12 @devmadhuu

Comment (smitajoshi12) [reply hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:98]: Done Changes Suggested by Zita.

image

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/db.json [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+     "capacity": 202114732032,
+     "used": 16384,
+     "remaining": 182447632384
```
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/db.json:9]:
```suggestion
      "capacity": 202114732032,
      "used": 16384,
      "remaining": 182447632384
```

File: hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx [hunk 4]
```diff
@@ -146,7 +144,7 @@ export class Overview extends React.Component<Record<string, object>, IOverviewS
     const duLink = '/DiskUsage';
     const containersElement = missingContainersCount > 0 ? (
       <span>
-        <Tooltip placement='bottom' title={`${missingContainersCount} ${containersTooltip}`}>
+        <Tooltip placement='bottom' title={missingContainersCount > 1000 ? `1000+ Containers are missing, For more information, go to the Containers page.` : `${missingContainersCount} ${containersTooltip}`}>
```
Comment (smengcl) [inline hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/overview/overview.tsx:147]: Need to use period `.` instead of comma `,` here.

```suggestion
        <Tooltip placement='bottom' title={missingContainersCount > 1000 ? `1000+ Containers are missing. For more information, go to the Containers page.` : `${missingContainersCount} ${containersTooltip}`}>
```

**Review by smengcl — APPROVED** [review]

**Comment by smengcl** [comment]
Thanks @smitajoshi12 for the improvement, and @dombizita for the review.
